### PR TITLE
[optin/puller]: get nem block timestamps workflow

### PR DESCRIPTION
**Pull request by gimre-xymcity**

### CODE REVIEW

File: optin/puller/tests/client/test_NemClient.py [hunk 1]
```diff
@@ -266,3 +277,20 @@ async def test_get_incoming_transactions_from_can_complete_in_multiple_remote_ca
 	] == transactions
 
 # endregion
+
+
+# region get_incoming_transactions_from
```
Comment (Jaguar0625) [inline optin/puller/tests/client/test_NemClient.py:282]: block_headers ? 
move up before free function (get_incoming_transactions_from) tests

File: optin/puller/puller/db/CompletedOptinDatabase.py [hunk 2]
```diff
@@ -45,6 +45,10 @@ def create_tables(self):
 			optin_id integer,
 			FOREIGN KEY (optin_id) REFERENCES optin_id(id)
 		)''')  # address cannot be unique because merges are supported
+		cursor.execute('''CREATE TABLE IF NOT EXISTS nem_block_timestamps (
+			height PRIMARY KEY,
```
Comment (Jaguar0625) [inline optin/puller/puller/db/CompletedOptinDatabase.py:49]: integer ?

File: optin/puller/puller/db/CompletedOptinDatabase.py [hunk 3]
```diff
@@ -45,6 +45,10 @@ def create_tables(self):
 			optin_id integer,
 			FOREIGN KEY (optin_id) REFERENCES optin_id(id)
 		)''')  # address cannot be unique because merges are supported
+		cursor.execute('''CREATE TABLE IF NOT EXISTS nem_block_timestamps (
```
Comment (cryptoBeliever) [inline optin/puller/puller/db/CompletedOptinDatabase.py:48]: We need NEM timestamps also for InProgressOptinDatabase.optin_error and optin_request. did you plan to add it there too?

File: optin/puller/workflows/download_nem_timestamps.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	args = parse_args()
+	nem_client = NemClient(args.nem_node)
+
+	with Databases(args.database_directory) as databases:
+		databases.completed.create_tables()
+
+		heights = databases.completed.transaction_heights()
+		existing_block_heights = set(map(lambda tuple: tuple[0], databases.completed.block_timestamps()))
+
+	timestamps = {}
```
Comment (Jaguar0625) [inline optin/puller/workflows/download_nem_timestamps.py:27]: height_timestamp_map ?

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**